connection.go

```diff
@@ -48,6 +48,22 @@ type mysqlConn struct {
 	finished chan<- struct{}
 	canceled atomicError // set non-nil if conn is canceled
 	closed   atomicBool  // set when conn is closed, before closech is closed
+
+	// for killing query after timeout
+	id  int
+	d   MySQLDriver
+	dsn string
+}
+
+func (mc *mysqlConn) kill() error {
+	conn, err := mc.d.Open(mc.dsn)
```

Comment:
Don't creating new connection pool!  Create mysqlConn directly.
Addtionally, opening new connection should be have short timeout.

Reply:
I agree about timeout but didn't catch your idea about connection pool. Here I'm using [MySQLDriver.Open](https://github.com/go-sql-driver/mysql/blob/master/driver.go#L59) method, which returns driver.Conn and error (underlying type of  driver.Conn is *mysqlConn)

Reply:
I'm sorry, I was wrong about the pool.

Reply:
#705 should be used instead of Driver.Open.

Reply:
Ok, will use after it merged to master